src/common/base-store.ts

```diff
@@ -168,7 +168,7 @@ export abstract class BaseStore<T> extends Singleton {
 
   protected async onModelChange(model: T) {
     if (ipcMain) {
-      this.saveToFile(model); // save config file
+      await this.saveToFile(model); // save config file
```

Comment:
I feel like this isn't the solution here. For me, it looks that the real issue is that `onModelChange` is called multiple times. Not sure though.

Reply:
If writing to the storage file is `sync` why it could be a an issue?

Reply:
> If writing to the storage file is `sync` why it could be a an issue?

`onModelChange` can be triggered multiple times, if we don't `await` then the order of execution of `sync` writes might happen out of order.

Reply:
Ok. What if remove `async` from `onModelChange`? And probable from `saveToFile`.

Reply:
> Ok. What if remove `async` from `onModelChange`? And probably from `saveToFile`.

It works as well but it modifies api behaviour (this class is exposed via extension api).

Reply:
> It works as well but it modifies api behaviour (this class is exposed via extension api).

No, making a previously `async` function `sync` is not a breaking API change in JS. Since you can `await` any value.

Reply:
I would say that making both `onModelChange` and `saveToFile` sync would be the best.

Reply:
If they override them, they can use `async` if they want:

https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ1oCMwCcAUAlAFzQBu8AlgCYrobTDwB2E8IApgHQjwDmeAIgpM4iTthwCCAbjoBfVAtShIMAEJgALtHYAPTeyZUYCJGgyQAnk2BZchEgAUc8ALYUI7ADzlqAPlp6BmZWDm4+QWFoDU1xXClZDAU5IA

Reply:
Yes, you can override but in this case parent methods becomes `async` when you call inside `super.method()` (but we expect it to be `sync`)

Reply:
@Nokel81 yes, probably you're right. 
Initially I thought there could be some timing issues, but maybe not..
This code produces expected result:

```tsx
class Foo {
  constructor(){
    this.bar();
  }

  bar(): void {
    console.log("in Foo.bar");
  }
}

class Bat extends Foo {
  async bar(): Promise<void> {
    await new Promise(resolve => setTimeout(resolve, 500));
    super.bar();
    console.log("in Bat.bar");
  }
}

new Bat();
```

**OUTPUT:**
```
[LOG]: "in Foo.bar" 
[LOG]: "in Bat.bar" 
```

Reply:
Outcome of this discussion: we need to remove async from these functions so that reaction will be sync.

@ixrock @Nokel81 right?

Reply:
That is what I believe yes.

Reply:
@jakolehm yes, it could be best solution I guess.

Reply:
I have been testing different things and I'm quite certain now that this PR mainly just adds tests for BaseStore, nothing actually changed in the execution order. I still removed `async` because it's quite misleading in this context.